verifiable-db/src/query/universal_circuit/universal_query_circuit.rs

```diff
@@ -365,45 +434,75 @@ where
     [(); MAX_NUM_RESULTS - 1]:,
     [(); MAX_NUM_COLUMNS + MAX_NUM_RESULT_OPS]:,
 {
-    /// Provide input values for universal circuit variant for queries with aggregation operations
-    pub(crate) fn new_query_with_agg(
+    /// Provide input values for universal circuit variant for queries without aggregation operations
+    pub(crate) fn new_query_no_agg(
         column_cells: &RowCells,
         predicate_operations: &[BasicOperation],
         placeholders: &Placeholders,
         is_leaf: bool,
         query_bounds: &QueryBounds,
         results: &ResultStructure,
     ) -> Result<Self> {
-        Ok(UniversalCircuitInput::QueryWithAgg(
+        Ok(UniversalCircuitInput::QueryNoAgg(
             UniversalQueryCircuitInputs::new(
                 column_cells,
                 predicate_operations,
                 placeholders,
                 is_leaf,
                 query_bounds,
                 results,
+                false,
             )?,
         ))
     }
-    /// Provide input values for universal circuit variant for queries without aggregation operations
-    pub(crate) fn new_query_no_agg(
-        column_cells: &RowCells,
+
+    pub(crate) fn ids_for_placeholder_hash(
         predicate_operations: &[BasicOperation],
+        results: &ResultStructure,
         placeholders: &Placeholders,
-        is_leaf: bool,
         query_bounds: &QueryBounds,
-        results: &ResultStructure,
-    ) -> Result<Self> {
-        Ok(UniversalCircuitInput::QueryNoAgg(
-            UniversalQueryCircuitInputs::new(
-                column_cells,
-                predicate_operations,
-                placeholders,
-                is_leaf,
-                query_bounds,
-                results,
-            )?,
-        ))
+    ) -> Result<[PlaceholderId; 2 * (MAX_NUM_PREDICATE_OPS + MAX_NUM_RESULT_OPS)]> {
+        let row_cells = &RowCells::default();
+        Ok(match results.output_variant {
+            Output::Aggregation => {
+                let circuit = UniversalQueryCircuitInputs::<
+                    MAX_NUM_COLUMNS,
+                    MAX_NUM_PREDICATE_OPS,
+                    MAX_NUM_RESULT_OPS,
+                    MAX_NUM_RESULTS,
+                    AggOutputCircuit<MAX_NUM_RESULTS>,
+                >::new(
+                    row_cells,
+                    predicate_operations,
+                    placeholders,
+                    false, // doesn't matter for placeholder hash computation
```

Comment:
if it doesn't matter in both cases (agg or no agg), then why keeping these arguments in the first place ? :)

Reply:
Because in some other cases (e.g., when we want to instantiate a dummy row to pad to maximum number of rows processed in a circuit) those parameters are meaningful. Maybe I could use an `Option` so that we just provide `None` in cases where the value is not relevant?

Reply:
In commit https://github.com/Lagrange-Labs/mapreduce-plonky2/pull/417/commits/0666bb9ba327edf45bd1899b89d63cf95036a179,  I changed the code to avoid instantiating that data structure providing dummy values for these flags